p_steps/step_12_2_run_scri.R

```diff
@@ -0,0 +1,252 @@
+                           dataformat = "multi",
+                           sameexpopar = F, 
+                           data = scri_input) 
+  
+  # extract relevant results 
+  output.mod.1 <- as.data.frame(cbind(output.1$conf.int, output.1$coefficients[,c(1,3)])) %>%
+    rename(irr = `exp(coef)`,
+           lci = `lower .95`,
+           uci = `upper .95`,
+           yi = `coef`,
+           sei = `se(coef)`) %>%
+    select(irr, lci, uci, yi, sei) %>% 
+    mutate(var = rownames(.), 
+           analysis = "unadjusted") 
+  
+  # Adjusted core model -------------------------------------------------------------- 
+  # Note - calendar time adjustment is added through a variable called age in this function, as per Farrington et al 2019. 
+  # Note - lots of convergence issues with calendar time variable included here, unsure whether a feature of simulated data 
+  # reviewer - please confirm specification looks ok 
+  output.2 <- standardsccs(event ~ vd + age, 
+                           indiv = case, 
+                           astart = start, 
+                           aend = end-1,
+                           aevent = event,
+                           adrug = cbind(vd, vd1, vd1+1, vd3, vd2, vd2+1),
+                           aedrug = cbind(vd1-1, vd1, r1end, vd2-1, vd2, r2end), 
+                           agegrp = calendar_time, 
+                           dataformat = "multi",
+                           sameexpopar = c(F,F), 
+                           data = scri_input) 
+  
+  # extract relevant results 
+  output.mod.2 <- as.data.frame(cbind(output.2$conf.int, output.2$coefficients[,c(1,3)])) %>%
+    rename(irr = `exp(coef)`,
+           lci = `lower .95`,
+           uci = `upper .95`,
+           yi = `coef`,
+           sei = `se(coef)`) %>%
+    select(irr, lci, uci, yi, sei) %>% 
+    mutate(var = rownames(.), 
+           analysis = "adjusted") 
+  
+  # Format results  -------------------------------------------------------
+  # stack and format datasets 
+  output.total <- rbind(output.mod.1, output.mod.2) %>% 
+    mutate(label = case_when(var == "vd1" ~ "pre-exposure pindow", 
+                             var == "vd2" ~ "dose 1 day 0", 
+                             var == "vd3" ~ "dose 1 risk window", 
+                             var == "vd4" ~ "time between dose 1 and dose 2", 
+                             var == "vd5" ~ "dose 2 day 0", 
+                             var == "vd6" ~ "dose 2 risk window", 
+                             TRUE ~ var)) %>% 
+    select(-c(var))
+  
+  # add columns with vaccine and outcome type 
+  # slightly hacky solution here, suggestions for improvement v welcome!
+  output.total$vacctype <- scri_input$vaccinetype[[1]] 
+  output.total$eventtype <- scri_input$eventtype[[1]]
+  
+  output.total <- output.total %>% 
+    relocate(vacctype, eventtype, analysis, label, irr, lci, uci, sei, yi)
+  
+  return(output.total)
+  
```

Comment:
`output.total` now has no variable to denote data source (DAP). This should be added because we meta-analyse across DAPs so for the meta-analysis code I need to know the data source.
I also need a variable that denotes the population subset as we will include several stratified analyses and these need to be meta-analysed separately.
The proposed output table is described in _Appendix I - Table 4_ of the SAP. This one also includes _ncase_ and _nevent_ columns which I need for forest plot completeness. Not sure if these can be added easily here, or if I should add them manually based on the flowcharts myself.